# [\#14072 Issue](https://github.com/ethereum/solidity/issues/14072) `closed`: [Codegen] Potential stack corruption?
**Labels**: `bug :bug:`


#### <img src="https://avatars.githubusercontent.com/u/2388185?v=4" width="50">[bshastry](https://github.com/bshastry) opened issue at [2023-03-27 07:16](https://github.com/ethereum/solidity/issues/14072):

```
contract C {
  bytes30 s = 0;
  function f(int i) public {
    payable(this).transfer(0);
  }
  receive() external payable {
    s;
    bytes30 l;
    assert(l == s);
  }
}
// ----
// f(int256): 0 ->
```

reverts without optimization only. With optimization it passes

Repro

```
$ cp test.sol test/libsolidity/semanticTests
$ isoltest -t semanticTests/test
```

#### <img src="https://avatars.githubusercontent.com/u/2388185?v=4" width="50">[bshastry](https://github.com/bshastry) commented at [2023-03-27 07:25](https://github.com/ethereum/solidity/issues/14072#issuecomment-1484638402):

It looks like an assertion failure but a panic code (https://docs.soliditylang.org/en/latest/control-structures.html#panic-via-assert-and-error-via-require) is not emitted. `transfer` may be reverting and not propagating the panic due to failed assertion.

#### <img src="https://avatars.githubusercontent.com/u/2388185?v=4" width="50">[bshastry](https://github.com/bshastry) commented at [2023-03-27 09:28](https://github.com/ethereum/solidity/issues/14072#issuecomment-1484810396):

Closing because @chriseth strongly feels this has to do with gas throttling of the transfer call, and the storage read in `receive` being too expensive for unoptimized code.


-------------------------------------------------------------------------------



[Export of Github issue for [ethereum/solidity](https://github.com/ethereum/solidity). Generated on 2024.12.15 at 06:45:24.]
